### ISSUE TITLE
Train-Accuracy decrease to 0

### ISSUE THREAD

**Issue by livedimg** [xlvector/learning-dl#1]
when learning_rate = 0.0001 after batch 1300 Train-Accuracy=0.000000.
And this batch number is always changing when next time training.
2016-07-01 12:27:17,341 Epoch[0] Batch [50] Speed: 181.20 samples/sec Train-Accuracy=0.000000
2016-07-01 12:27:26,751 Epoch[0] Batch [100] Speed: 170.05 samples/sec Train-Accuracy=0.000000
2016-07-01 12:27:36,490 Epoch[0] Batch [150] Speed: 164.29 samples/sec Train-Accuracy=0.000000
2016-07-01 12:27:46,196 Epoch[0] Batch [200] Speed: 164.85 samples/sec Train-Accuracy=0.000400
2016-07-01 12:27:56,264 Epoch[0] Batch [250] Speed: 158.92 samples/sec Train-Accuracy=0.001600
2016-07-01 12:28:05,724 Epoch[0] Batch [300] Speed: 169.14 samples/sec Train-Accuracy=0.004000
2016-07-01 12:28:15,906 Epoch[0] Batch [350] Speed: 157.14 samples/sec Train-Accuracy=0.001600
2016-07-01 12:28:25,906 Epoch[0] Batch [400] Speed: 160.00 samples/sec Train-Accuracy=0.000400
2016-07-01 12:28:35,385 Epoch[0] Batch [450] Speed: 168.79 samples/sec Train-Accuracy=0.006800
2016-07-01 12:28:45,341 Epoch[0] Batch [500] Speed: 160.71 samples/sec Train-Accuracy=0.023200
2016-07-01 12:28:54,888 Epoch[0] Batch [550] Speed: 167.60 samples/sec Train-Accuracy=0.076000
2016-07-01 12:29:04,616 Epoch[0] Batch [600] Speed: 164.47 samples/sec Train-Accuracy=0.122800
2016-07-01 12:29:14,225 Epoch[0] Batch [650] Speed: 166.50 samples/sec Train-Accuracy=0.161600
2016-07-01 12:29:23,789 Epoch[0] Batch [700] Speed: 167.30 samples/sec Train-Accuracy=0.172400
2016-07-01 12:29:33,543 Epoch[0] Batch [750] Speed: 164.03 samples/sec Train-Accuracy=0.175200
2016-07-01 12:29:43,938 Epoch[0] Batch [800] Speed: 153.93 samples/sec Train-Accuracy=0.242800
2016-07-01 12:29:53,821 Epoch[0] Batch [850] Speed: 161.88 samples/sec Train-Accuracy=0.264800
2016-07-01 12:30:03,501 Epoch[0] Batch [900] Speed: 165.30 samples/sec Train-Accuracy=0.282400
2016-07-01 12:30:12,893 Epoch[0] Batch [950] Speed: 170.37 samples/sec Train-Accuracy=0.305200
2016-07-01 12:30:22,977 Epoch[0] Batch [1000] Speed: 158.66 samples/sec Train-Accuracy=0.304000
2016-07-01 12:30:32,574 Epoch[0] Batch [1050] Speed: 166.72 samples/sec Train-Accuracy=0.265600
2016-07-01 12:30:42,185 Epoch[0] Batch [1100] Speed: 166.48 samples/sec Train-Accuracy=0.000400
2016-07-01 12:30:51,542 Epoch[0] Batch [1150] Speed: 170.99 samples/sec Train-Accuracy=0.004800
2016-07-01 12:31:01,694 Epoch[0] Batch [1200] Speed: 157.61 samples/sec Train-Accuracy=0.000000
2016-07-01 12:31:11,258 Epoch[0] Batch [1250] Speed: 167.30 samples/sec Train-Accuracy=0.000400
2016-07-01 12:31:20,939 Epoch[0] Batch [1300] Speed: 165.27 samples/sec Train-Accuracy=0.000000
2016-07-01 12:31:31,098 Epoch[0] Batch [1350] Speed: 157.51 samples/sec Train-Accuracy=0.000000
2016-07-01 12:31:40,550 Epoch[0] Batch [1400] Speed: 169.26 samples/sec Train-Accuracy=0.000000
2016-07-01 12:31:50,023 Epoch[0] Batch [1450] Speed: 168.91 samples/sec Train-Accuracy=0.000000
2016-07-01 12:31:59,475 Epoch[0] Batch [1500] Speed: 169.27 samples/sec Train-Accuracy=0.000400
2016-07-01 12:32:08,882 Epoch[0] Batch [1550] Speed: 170.09 samples/sec Train-Accuracy=0.000400
2016-07-01 12:32:18,604 Epoch[0] Batch [1600] Speed: 164.58 samples/sec Train-Accuracy=0.000000

And I found ,modify the learning_rate = 0.00005, it became normal
2016-07-02 18:00:39,762 Epoch[6] Batch [3328]   Speed: 131.35 samples/sec       Train-Accuracy=0.849609
2016-07-02 18:00:43,708 Epoch[6] Batch [3392]   Speed: 129.74 samples/sec       Train-Accuracy=0.869141
2016-07-02 18:00:47,561 Epoch[6] Batch [3456]   Speed: 132.91 samples/sec       Train-Accuracy=0.837891
2016-07-02 18:00:51,499 Epoch[6] Batch [3520]   Speed: 130.01 samples/sec       Train-Accuracy=0.833984
2016-07-02 18:00:55,285 Epoch[6] Batch [3584]   Speed: 135.25 samples/sec       Train-Accuracy=0.826172
2016-07-02 18:00:59,653 Epoch[6] Batch [3648]   Speed: 117.22 samples/sec       Train-Accuracy=0.849609
2016-07-02 18:01:03,587 Epoch[6] Batch [3712]   Speed: 130.16 samples/sec       Train-Accuracy=0.818359
2016-07-02 18:01:07,906 Epoch[6] Batch [3776]   Speed: 118.54 samples/sec       Train-Accuracy=0.857422
2016-07-02 18:01:12,106 Epoch[6] Batch [3840]   Speed: 121.90 samples/sec       Train-Accuracy=0.847656
2016-07-02 18:01:16,395 Epoch[6] Batch [3904]   Speed: 119.40 samples/sec       Train-Accuracy=0.826172
2016-07-02 18:01:20,811 Epoch[6] Batch [3968]   Speed: 115.94 samples/sec       Train-Accuracy=0.847656
2016-07-02 18:01:24,685 Epoch[6] Batch [4032]   Speed: 132.17 samples/sec       Train-Accuracy=0.871094
2016-07-02 18:01:28,553 Epoch[6] Batch [4096]   Speed: 132.34 samples/sec       Train-Accuracy=0.816406
2016-07-02 18:01:32,780 Epoch[6] Batch [4160]   Speed: 121.15 samples/sec       Train-Accuracy=0.833984
2016-07-02 18:01:37,451 Epoch[6] Batch [4224]   Speed: 109.59 samples/sec       Train-Accuracy=0.869141
2016-07-02 18:01:41,288 Epoch[6] Batch [4288]   Speed: 133.44 samples/sec       Train-Accuracy=0.869141
2016-07-02 18:01:45,516 Epoch[6] Batch [4352]   Speed: 121.12 samples/sec       Train-Accuracy=0.888672
2016-07-02 18:01:50,056 Epoch[6] Batch [4416]   Speed: 112.78 samples/sec       Train-Accuracy=0.867188
2016-07-02 18:01:54,136 Epoch[6] Batch [4480]   Speed: 125.50 samples/sec       Train-Accuracy=0.869141
2016-07-02 18:01:57,712 Epoch[6] Batch [4544]   Speed: 143.15 samples/sec       Train-Accuracy=0.859375
2016-07-02 18:02:01,307 Epoch[6] Batch [4608]   Speed: 142.42 samples/sec       Train-Accuracy=0.828125
2016-07-02 18:02:04,951 Epoch[6] Batch [4672]   Speed: 140.53 samples/sec       Train-Accuracy=0.839844
2016-07-02 18:02:08,536 Epoch[6] Batch [4736]   Speed: 142.84 samples/sec       Train-Accuracy=0.820312
2016-07-02 18:02:12,125 Epoch[6] Batch [4800]   Speed: 142.65 samples/sec       Train-Accuracy=0.843750
2016-07-02 18:02:15,527 Epoch[6] Batch [4864]   Speed: 150.49 samples/sec       Train-Accuracy=0.832031
2016-07-02 18:02:19,359 Epoch[6] Batch [4928]   Speed: 133.64 samples/sec       Train-Accuracy=0.855469
2016-07-02 18:02:22,822 Epoch[6] Batch [4992]   Speed: 147.85 samples/sec       Train-Accuracy=0.886719
2016-07-02 18:02:26,696 Epoch[6] Batch [5056]   Speed: 132.17 samples/sec       Train-Accuracy=0.859375
2016-07-02 18:02:30,049 Epoch[6] Batch [5120]   Speed: 152.71 samples/sec       Train-Accuracy=0.841797
2016-07-02 18:02:33,966 Epoch[6] Batch [5184]   Speed: 130.71 samples/sec       Train-Accuracy=0.835938
2016-07-02 18:02:37,485 Epoch[6] Batch [5248]   Speed: 145.49 samples/sec       Train-Accuracy=0.835938
2016-07-02 18:02:41,262 Epoch[6] Batch [5312]   Speed: 135.59 samples/sec       Train-Accuracy=0.863281
2016-07-02 18:02:45,085 Epoch[6] Batch [5376]   Speed: 133.91 samples/sec       Train-Accuracy=0.847656
2016-07-02 18:02:49,036 Epoch[6] Batch [5440]   Speed: 129.62 samples/sec       Train-Accuracy=0.851562
2016-07-02 18:02:52,920 Epoch[6] Batch [5504]   Speed: 131.82 samples/sec       Train-Accuracy=0.828125
2016-07-02 18:02:57,258 Epoch[6] Batch [5568]   Speed: 118.03 samples/sec       Train-Accuracy=0.816406
2016-07-02 18:03:01,111 Epoch[6] Batch [5632]   Speed: 132.88 samples/sec       Train-Accuracy=0.861328
2016-07-02 18:03:05,454 Epoch[6] Batch [5696]   Speed: 117.90 samples/sec       Train-Accuracy=0.818359
2016-07-02 18:03:09,288 Epoch[6] Batch [5760]   Speed: 133.54 samples/sec       Train-Accuracy=0.832031
2016-07-02 18:03:13,130 Epoch[6] Batch [5824]   Speed: 133.28 samples/sec       Train-Accuracy=0.800781
2016-07-02 18:03:16,970 Epoch[6] Batch [5888]   Speed: 133.34 samples/sec       Train-Accuracy=0.818359
2016-07-02 18:03:20,859 Epoch[6] Batch [5952]   Speed: 131.65 samples/sec       Train-Accuracy=0.845703

**Comment by xlvector** [comment]
you run the same program as me？did you change something？

**Comment by xlvector** [comment]
Thanks
